lib/builder.js

```diff
@@ -32,19 +34,27 @@ function summarize(node) {
 RSVP.EventTarget.mixin(Builder.prototype)
 
 Builder.prototype.build = function (willReadStringTree) {
+  if (this.canceled) {
+    return RSVP.Promise.reject(new Error('cannot build this builder, as it has been previously canceled'));
```

Comment:
I think its ok as an Error. My reasoning is: this is trully exceptional, and should be reported.